### DISCUSSION TITLE
MIDI  presets

### DISCUSSION THREAD

**Discussion by n3ptun3s** [SynthstromAudible/DelugeFirmware#387] [Ideas]
People with many external Synths would love this option:
Being able to save Presets in Midi Mode. 
So that you can load a named Midi Device with adjusted CCs values for the golden knobs.
I don't know though, if the present preset structure can handle this.

**Comment by m-m-adams** [comment]
This is a great idea! I'll probably look into it eventually but full support for anyone else who wants to do it

**Comment by croteaup** [comment]
I remember that midi presets was at some point part of the road map and I understand why. It wasn’t named “midi presets” tho.
+1

**Comment by Xchco** [comment]
Would love this. Squarp calls them "Definition Files" and they are great. They are also very easy to create and share. Basically it's a text file that anyone -- even with absolutely no coding experience -- can change or create. I'd recommended taking a look at how Squarp does it as reference.

**Comment by Xchco** [comment]
Here's the link to Squarp Pyramid's manual, Definition Files section:

https://squarp.net/legacy/pyramid/manual/definitionfiles/

**Reply by Xchco** [reply]
And here is Squarp community's "Definiton Files" library

https://squarp.community/c/def/11

**Comment by solaris76** [comment]
I would like this also, but why not go further and allow the saving of the note (pattern) information too? i.e. the ability to save/browse clip/pattern presets?

**Reply by Xchco** [reply]
I think that that is a different concept/paradigm, if I understand you correctly. Why have a pattern associated with a midi preset? Think of the midi preset as just a synth-specific midi controller template within the Deluge (that you can load within any project). You load, say, the midi preset for the Novation Peak onto a row/track -- now everything is mapped out (filter, pitch bend, etc.) to the Peak's specific CC numbers/parameters, so now the golden knobs can be assigned to those parameters by name and not by number. You can have a specific midi channel assigned as well to that midi preset, and so on. If it's a sampler or drum machine and each instrument is not a CC value but a note (the kick, snare, etc.), you can map that as well onto the midi preset file so that note gives you that instruments name on the screen when you press it.
Each preset has the gear's name or an abbreviated form of it. Now you can scroll between midi preset names as you would synth presets.

This is Squarp's way of doing and it is excellent. Extremely simple and quick to create files and - very importantly - share them.

**Reply by solaris76** [reply]
I get what you're saying, and agree with all, but I'm suggesting it saves the pattern/notes with it as well. If you don't want that then just clear the note data. It's essentially the same thing (and one piece of functionality to be developed), i.e. a track/clip preset.

**Reply by Xchco** [reply]
Well, for me personally I would rather a blank slate, but, of course, there is no right or wrong in this case, only preference.
Either way would be a great feature which I hope they will add!

Cheers

**Comment by m-m-adams** [comment]
This is completed - part of 1.1

**Reply by Xchco** [reply]
Awesome! Will download now

Thank you!

**Reply by jtdaugherty** [reply]
@m-m-adams could you provide a pointer to where this is described in [the change list](https://github.com/SynthstromAudible/DelugeFirmware/blob/release/1.1/docs/community_features.md)? I don't see anything there that resembles what was discussed here, as best I can tell. (I also don't see a link in the changes to this discussion.) Thanks!
